mewarpx/mewarpx/diags_store/diag_base.py

```diff
@@ -208,7 +214,7 @@ def text_diag(self):
                 else:
                     step_rate = 0
 
-                particle_step_rate = live_parts * step_rate
+                particle_step_rate = (self.particle_steps_total - self.previous_particle_steps_total) / step_rate
```

Comment:
You want to use `wall_time` as the denominator here, probably within the conditional just above.

Reply:
Just so I'm clear, what does within the conditional men exactly? I would guess it means inside `if wall_time > 0:` and otherwise maybe `particle_step_rate` is 0?

Reply:
It's just setting a reasonable-looking value for the calculation at the beginning of the run before any time has passed, to avoid division by zero.